### DISCUSSION TITLE
System requirements

### DISCUSSION THREAD

**Discussion by badprogrammer42** [headlesshq/headlessmc#187] [Q&A]
Hey,
I would like to ask, what are the minimum system requirements for HeadlessMC to run in headless mode?

**Comment by 3arthqu4ke** [comment]
Honestly, not sure, I really gotta make a benchmark sometime.
The lowest you can probably get currently is by using `hmc.assets.dummy=true`, and the [headlessmc-optimizations](https://github.com/3arthqu4ke/hmc-optimizations). With that I was able to join a Singleplayer world with 1GB of RAM in 1.21.1, but I think you could potentially go lower. You can also turn down your settings and servers instead of SinglePlayer worlds should consume even less.

**Reply by badprogrammer42** [reply]
Thank you for your quick response. I will try it out like you said with the `hmc.assets.dummy=true`.
I run the version 1.8.9 so sadly there is no headlessmc-optimizations for this.

**Reply by badprogrammer42** [reply]
btw can you please tell me how I can set the memory limit up when I am running in in-memory mode?

**Reply by 3arthqu4ke** [reply]
In-Memory just means that it will run in the same jvm as headlessmc, so whatever memory you give headlessmc, minecraft will get too, so you start headlessmc like this `java -Xmx2G -jar headlessmc-launcher-wrapper.jar (Xmx2G for 2GB or RAM)

**Reply by badprogrammer42** [reply]
ok thank you
